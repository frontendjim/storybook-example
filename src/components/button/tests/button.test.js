import React from 'react';
import { render } from '@testing-library/react';

import Button from '..';

describe('<Button />', () => {
  const mockProps = {
    label: 'Test label',
    onClick: () => null,
  };

  it('should render as expended', () => {
    const { asFragment } = render(<Button {...mockProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render an icon when requested', () => {
    const { container } = render(<Button {...mockProps} icon="plus" />);

    const iconEl = container.firstChild.querySelector('span');
    expect(iconEl).not.toBeNull();
  });
});
