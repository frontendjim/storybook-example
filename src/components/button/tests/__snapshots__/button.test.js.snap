// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render as expended 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm cFExar"
    type="primary"
  >
    Test label
  </button>
</DocumentFragment>
`;
